middle_end/flambda/simplify/simplify_switch_expr.ml

```diff
@@ -38,23 +38,24 @@ let rebuild_switch ~simplify_let dacc ~arms ~scrutinee ~scrutinee_ty uacc
             else
               let cont = Apply_cont.continuation action in
               match UE.find_continuation (UA.uenv uacc) cont with
-              | Linearly_used_and_inlinable { arity = _; handler;
-                  free_names_of_handler = _; params; cost_metrics_of_handler = _ } ->
+              | Linearly_used_and_inlinable { handler;
+                  free_names_of_handler = _; params;
+                  cost_metrics_of_handler = _ } ->
                 assert (List.length params = 0);
                 begin match Expr.descr handler with
                 | Apply_cont action -> Some action
                 | Let _ | Let_cont _ | Apply _
                 | Switch _ | Invalid _ -> Some action
                 end
-              | Other { arity = _; handler = Some handler; } ->
-                Continuation_handler.pattern_match handler
-                  ~f:(fun params ~handler ->
-                    assert (List.length params = 0);
-                    match Expr.descr handler with
-                    | Apply_cont action -> Some action
-                    | Let _ | Let_cont _ | Apply _
-                    | Switch _ | Invalid _ -> Some action)
-              | Other _ -> Some action
+              | Non_inlinable { params; handler = Known handler; } ->
```

Comment:
This pattern could now be merged with the one above since they bind the exact same set of variables (`params` and `handler`, this would allow to share the code for the two branches, which are also exactly the same, and make it clearer that we do the same thing in both cases.

Reply:
This isn't possible after another change I've just made (see next comment) but I have factored this code out into a function now that will always be statically allocated.